add cross region vpc peering

mteiste opened the pull request:
fix issue #56

rikonen commented on resource_vpc_peering_connection.go:
```diff
@@ -41,6 +41,18 @@ func resourceVPCPeeringConnection() *schema.Resource {
 				Type:        schema.TypeString,
 			},
+			"peer_region": {
+				Description: "AWS region of the peered VPC (if not in the same region)",
```
If not in the same region as what.

mteiste replied:
fixed

rikonen commented on resource_vpc_peering_connection.go:
```diff
@@ -41,6 +41,18 @@ func resourceVPCPeeringConnection() *schema.Resource {
+					if new != "" {
+						return false
+					}
+					return true
```
`return new == ""` would work as well I suppose

mteiste replied:
fixed

rikonen commented on resource_vpc_peering_connection.go:
```diff
@@ -58,15 +70,34 @@ func resourceVPCPeeringConnection() *schema.Resource {
+		err error
 	)
 
+	if peerRegion != "" {
```
Only having single `Create` in the API client and something like

```
apiPeerRegion := nil
if peerRegion != "" {
    apiPeerRegion = &peerRegion
}
pc, err := client.VPCPeeringConnections.Create(...)
```

would seem more convenient.

mteiste commented:
manually tested
- created vpc and one peering in the same region and one peering in an other region
- applied and verified from Aiven console that the vpc and peerings were created
- refreshed
- tested plan (to test that the same region peering will stay unchanged even though the refresh will fill the peer_region
- removed peerings from the configuration
- applied and verified from Aiven console that the peerings were deleted
- removed vpc from configuratoin
- applied and verified from Aiven console that the vpc status was changed to deleted
